pkg/api/metrics.go

```diff
@@ -28,6 +29,12 @@ func (hs *HTTPServer) handleQueryMetricsError(err error) *response.NormalRespons
 	if errors.Is(err, plugincontext.ErrPluginNotFound) {
 		return response.Error(http.StatusNotFound, "Plugin not found", err)
 	}
+	if errors.Is(err, plugins.ErrPluginNotRegistered) {
+		return response.Err(err)
+	}
+	if errors.Is(err, plugins.ErrMethodNotImplemented) {
+		return response.Err(err)
+	}
```

Comment:
If you make `plugins.ErrPluginNotRegistered` and `plugins.ErrMethodNotImplemented` based on errutil package you can skip these additional checks since the `response.ErrOrFallback` below would take care of them